### PR TITLE
made api in backend and fetched in front-end, used middleware

### PR DESCRIPTION
**Pull request by a5h101**

### CODE REVIEW

**Comment by a5h101** [comment]
I have not deleted the all console logs and comments because I will need them for some time

File: Rentit-backend/app.js [hunk 1]
```diff
@@ -1,12 +1,55 @@
...
+    },
+  });
+};
+const getItemDetails = (req, res) => {
+  const item = items.find((e) => e.id == req.params.id);
+  // console.log(req);
+  if (req.params.id > items.length - 1)
+    return res.status(404).json({
+      status: 'fail',
+      message: 'imvalid ID',
```
Comment (PatelNeelMahesh) [inline Rentit-backend/app.js:39]: "invalid ID"

File: Rentit-frontend/src/components/items/items.jsx [hunk 2]
```diff
@@ -1,23 +1,43 @@
 import './items.css';
 import { useNavigate } from 'react-router-dom';
-import fakedata from '../../fakedata.json';
+import { useEffect, useState } from 'react';
+
 const Items = () => {
+  const [itemData, setItemData] = useState(null);
+
+  useEffect(() => {
+    const url = 'http://127.0.0.1:2000/api-rentit/v1/items';
```
Comment (PatelNeelMahesh) [inline Rentit-frontend/src/components/items/items.jsx:9]: domain url, api keys and other sensitive details should be stored in .evn files and access using process.env

**Review by PatelNeelMahesh — CHANGES_REQUESTED** [review]
You need to add node_modules in .gitignore file and provide description for the PR.
After making the changes you can merge pull request.

**Pull request merged**